src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -615,8 +625,585 @@ class VM_Version_StubGenerator: public StubCodeGenerator {
+  const int CPUID_INDEX_CEIL = 8;
+  // /usr/include/bits/platform/x86.h
+  enum
+  {
+    CPUID_INDEX_1 = 0,
+    CPUID_INDEX_7,
+    CPUID_INDEX_80000001,
+    CPUID_INDEX_D_ECX_1,
+    CPUID_INDEX_80000007,
+    CPUID_INDEX_80000008,
+    CPUID_INDEX_7_ECX_1,
+    CPUID_INDEX_19,
+    CPUID_INDEX_14_ECX_0
+  };
+  const int index_max = CPUID_INDEX_CEIL + 1;
+  enum { eax = 0, ebx, ecx, edx, reg_max };
+  unsigned active[index_max][reg_max] = { 0 };
+  const char *arg0 = ld_so_name();
+  pid_t f_child;
+  FILE *f = popen_r(arg0, &f_child);
+  if (!f)
+    vm_exit_during_initialization(err_msg("Cannot popen %s " ARG1 ": %m", arg0));
+  for (;;) {
+    char line[LINE_MAX];
+    char *s = fgets(line, sizeof(line), f);
+    if (!s)
+      break;
+    s = line;
+    // x86.cpu_features.features[0x0].active[0x2]=0x7ed83203
+    const char prefix[] = "x86.cpu_features.features[";
+    if (strncmp(s, prefix, sizeof(prefix) - 1) != 0)
+      continue;
+    s += sizeof(prefix) - 1;
+    unsigned long index = strtoul(s, &s, 0);
+    if (index >= index_max)
+      continue;
+    const char mid[] = "].active[";
+    if (strncmp(s, mid, sizeof(mid) - 1) != 0)
+      continue;
+    s += sizeof(mid) - 1;
+    unsigned long reg = strtoul(s, &s, 0);
+    if (reg >= reg_max)
+      continue;
+    if (s[0] != ']' || s[1] != '=')
+      continue;
+    s += 2;
+    unsigned long val = strtoul(s, &s, 0);
+    if (val > UINT_MAX)
+      continue;
+    if (s[0] != '\n' || s[1] != 0)
+      continue;
+    active[index][reg] = val;
+  }
+  if (ferror(f))
+    vm_exit_during_initialization(err_msg("Error reading popen-ed %s " ARG1 ": %m", arg0));
+  if (!feof(f))
+    vm_exit_during_initialization(err_msg("EOF not reached on popen-ed %s " ARG1, arg0));
+  pclose_r(arg0, f, f_child);
+#undef ARG1
+#endif // INCLUDE_LD_SO_LIST_DIAGNOSTICS
+
+  // glibc: sysdeps/x86/get-isa-level.h:
+  // glibc: if (CPU_FEATURE_USABLE_P (cpu_features, CMOV)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, CX8)
+  // glibc:     && CPU_FEATURE_CPU_P (cpu_features, FPU)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, FXSR)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, MMX)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE)
+  // glibc:     && CPU_FEATURE_USABLE_P (cpu_features, SSE2))
+  // glibc:     isa_level = GNU_PROPERTY_X86_ISA_1_BASELINE;
+  if ((_features & CPU_CMOV) &&
+      (_features & CPU_CX8) &&
+      // FPU is always present on i686+: (_features & CPU_FPU) &&
+      (_features & CPU_SSE2))
+    // These cannot be disabled by GLIBC_TUNABLES.
+    if (excessive_CPU & (CPU_FXSR | CPU_MMX | CPU_SSE)) {
+      assert(!(excessive_CPU & CPU_SSE2), "CPU_SSE2 in both _features and excessive_CPU cannot happen");
+      // FIXME: The choice should be based on glibc impact, not the feature age.
+      // CX8 is i586+, CMOV is i686+ 1995+, SSE2 is 2000+
+      excessive_CPU |= CPU_SSE2;
+    }
+    if ((_features & CPU_FXSR) &&
```

Comment:
My compiler started to complain on this, and it looks valid:

```
src/hotspot/cpu/x86/vm_version_x86.cpp: In static member function 'static void VM_Version::glibc_not_using(uint64_t, uint64_t)':
src/hotspot/cpu/x86/vm_version_x86.cpp:1118:3: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
   if ((_features & CPU_CMOV) &&
   ^~
src/hotspot/cpu/x86/vm_version_x86.cpp:1129:5: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
     if ((_features & CPU_FXSR) &&
     ^~
cc1plus: all warnings being treated as errors
```

@jankratochvil it indeed looks like a missing bracket, is it a bug?

Reply:
Yes, sorry, #94.